sycl/doc/extensions/SPIRV/SPV_INTEL_arbitrary_precision_fixed_point.asciidoc

```diff
@@ -0,0 +1,470 @@
+OpFixedSinCosPiINTEL
+OpFixedLogINTEL
+OpFixedExpINTEL
+----
+
+== Token Number Assignments
+[width="40%"]
+[cols="70%,30%"]
+[grid="rows"]
+|====
+|`ArbitraryPrecisionFixedPointINTEL`    | 5922
+|`OpFixedSqrtINTEL`                     | 5923
+|`OpFixedRecipINTEL`                    | 5924
+|`OpFixedRsqrtINTEL`                    | 5925
+|`OpFixedSinINTEL`                      | 5926
+|`OpFixedCosINTEL`                      | 5927
+|`OpFixedSinCosINTEL`                   | 5928
+|`OpFixedSinPiINTEL`                    | 5929
+|`OpFixedCosPiINTEL`                    | 5930
+|`OpFixedSinCosPiINTEL`                 | 5931
+|`OpFixedLogINTEL`                      | 5932
+|`OpFixedExpINTEL`                      | 5933
+|====
+
+
+== Modifications to the SPIR-V Specification Version 1.5
+
+After Section 3.16, add a new section "3.16b Quantization Modes" as follows:
+
+=== Quantization Modes
+[options="header"]
+[width="80%"]
+[cols="15%,20%,65%"]
+|====
+| Value ^| Mode        ^| Behavior
+| 0      | TRN          | Truncate towards -Inf
+| 1      | TRN_ZERO     | Truncate towards 0
+| 2      | RND          | Round towards +Inf
+| 3      | RND_ZERO     | Round towards 0
+| 4      | RND_INF      | Round towards +/-Inf
+| 5      | RND_MIN_INF  | Round towards -Inf
+| 6      | RND_CONV     | Round towards even
+| 7      | RND_CONV_ODD | Round towards odd
+|====
+
+After Section 3.16, add a new section "3.16c Overflow Modes" as follows:
+
+=== Overflow Modes
+[options="header"]
+[width="80%"]
+[cols="15%,20%,65%"]
+|====
+| Value ^| Mode     ^| Behavior
+| 0      | WRAP      | Drop the bits to the left of the MSB
+| 1      | SAT       | Saturate to the closest of MIN or MAX
+| 2      | SAT_ZERO  | Set to 0 on overflow
+| 3      | SAT_SYM   | For unsigned, treat as SAT. +
+
+For signed: a positive overflow will saturate at the maximum positive value, whereas a negative overflow will saturate to the negation of the maximum positive value, as opposed to the most negative value.
+|====
+
+=== Capability
+Modify Section 3.31, *Capability*, adding a row to the Capability table:
+
+[options="header"]
+|====
+2+^| Capability ^| Implicitly Declares
+| 5922 | *ArbitraryPrecisionFixedPointINTEL* +
+
+Enables arbitrary precision fixed-point math instructions. +
+
+Note that this fixed-point type does not hold information about the sign in the type, but the information is rather encoded into the operations performed on this type.
+| *INTELArbitraryPrecisionIntegers*, *Kernel*
+|====
+
+=== Instructions
+
+In Section 3.32.13, *Arithmetic Instructions*, add the following instructions:
+
+[cols="10", width="100%"]
+|=====
+9+<|*OpFixedSqrtINTEL* +
+
+An `OpTypeInt` representing an arbitrary precision fixed point number (ac_fixed) is passed in as the _Input_ and the square root of the value is returned in _Result_.
+The behavior of this function is undefined for input values < 0.
+
+_Result Type_ is an `OpTypeInt` of width _rW_ and is the type of _Result_.
```

Comment:
I'm not sure if I asked this before, sorry if this is a duplicate: am I right that the intent is to allow integers of different width to be passed as `Input` and `Result` arguments, right?

Do we need to describe some limitations of `I` and `rI` literals? Something  like they can't contain values higher than number of bits in corresponding argument and that they are treated as unsigned values (or just can't be negative)

Reply:
Actually, this is not true. You can move the decimal beyond the size (in either direction) of the data type. For example, if you had a 2 bit data, and you set the value of I as follows, you can represent the following numbers: 
```
4: -32, -16, 0, 16
0:  -2,  -1, 0,  1
-4: 1b'0.111111, -1b'0.111110, 1b'0.000000, 1b'0.000001
```